sympy/solvers/tests/test_solveset.py

```diff
@@ -1749,3 +1737,164 @@ def test_issue_14454():
+
+    # coverage test
+    C1, C2 = symbols('C1 C2')
+    f = Function('f')
+    assert solveset_real(C1 + C2/x**2 - exp(-f(x)), f(x)) == \
+        Intersection(S.Reals, FiniteSet(-log(C1 + C2/x**2)))
+    y = symbols('y', positive=True)
+    assert solveset_real(x**2 - y**2/exp(x), y) == \
+        Intersection(S.Reals, FiniteSet(-sqrt(x**2*exp(x)), sqrt(x**2*exp(x))))
+    p = Symbol('p', positive=True)
+    assert solveset_real((1/p + 1)**(p + 1), p) == EmptySet()
+
+
+@XFAIL
+def _expo_solver_complex():
+    from sympy.abc import x
+    from sympy import Dummy
+    n = Dummy('n')
+
+    assert solveset_complex(2**x + 4**x, x) == \
+        imageset(Lambda(n, I*(2*n*pi + pi)/log(2)), S.Integers)
+    assert solveset_complex(x**z*y**z - 2, z) == \
+        FiniteSet(log(2)/(log(x) + log(y)))
+    assert solveset_complex(4**(x/2) - 2**(x/3), x) == \
+        imageset(Lambda(n, 3*n*I*pi/log(2)), S.Integers)
+    assert solveset(2**x + 32, x) == \
+        imageset(Lambda(n, (I*(2*n*pi + pi) + 5*log(2))/log(2)), S.Integers)
+    assert solveset(2**x + 4**x + 8**x, x) == \
+        Union(imageset(Lambda(n, I*(2*n*pi - 2*pi/3)/log(2)), S.Integers),
+              imageset(Lambda(n, I*(2*n*pi + 2*pi/3)/log(2)), S.Integers))
+    assert solveset(4**(x + 1) + 4**(x + 2) + 4**(x - 1) - 3**(x + 2) - 3**(x + 3), x) == \
+        imageset(Lambda(n, (2*n*I*pi - 4*log(2) + 2*log(3))/(-2*log(2) + log(3))),
+                 S.Integers)
+    assert solveset_complex((2**exp(y**2/x) + 2)/(x**2 + 15), y) == \
+        FiniteSet(-sqrt(x)*sqrt(-log(log(2)) + log(log(2) + 2*n*I*pi)),
+                  sqrt(x)*sqrt(-log(log(2)) + log(log(2) + 2*n*I*pi)))
+
+
+def test_expo_conditionset():
+    from sympy.abc import x, y
+
+    f1 = (exp(x) + 1)**x - 2
+    f2 = (x + 2)**y*x - 3
+    f3 = 2**x - exp(x) - 3
+    f4 = log(x) - exp(x)
+
+    assert solveset(f1, x, S.Reals) == \
+        ConditionSet(x, Eq((exp(x) + 1)**x - 2, 0), S.Reals)
+    assert solveset(f2, x, S.Reals) == ConditionSet(x, Eq(x*(x + 2)**y - 3, 0), S.Reals)
+    assert solveset(f3, x, S.Reals) == ConditionSet(x, Eq(2**x - exp(x) - 3, 0), S.Reals)
+    assert solveset(f4, x, S.Reals) == ConditionSet(x, Eq(-exp(x) + log(x), 0), S.Reals)
+
+
+def test_expo_symbols():
+    x, y, z = symbols('x y z', positive=True)
+    from sympy import simplify
+
+    assert solveset(z**x - y, x, S.Reals) == Intersection(
+        S.Reals, FiniteSet(log(y)/log(z)))
+
+    w = symbols('w', integer=True)
+    f1 = 2*x**w - 4*y**w
+    f2 = (x/y)**w - 2
+    ans1 = solveset(f1, w, S.Reals)
+    ans2 = solveset(f2, w, S.Reals)
+    assert ans1 == simplify(ans2)
+
+    assert solveset(x**x, x, S.Reals) == S.EmptySet
+    assert solveset(x**y - 1, y, S.Reals) == FiniteSet(0)
+    assert solveset(exp(x/y)*exp(-z/y) - 2, y, S.Reals) == FiniteSet(
+        (x - z)/log(2)) - FiniteSet(0)
+
+
+@XFAIL
+def test_issue_10864():
+    assert solveset(x**(y*z) - x, x, S.Reals) == FiniteSet(1)
```

Comment:
this test was previously solved as by `_expo_solver` because `_check_expo` returned True, since then it checked if the equation had power terms, but now `_check_expo` is improved and it checks for `symbol` in the exponent then only returns True.